tests/bson/bson-binary-set-state-001.phpt

```diff
@@ -0,0 +1,56 @@
+--TEST--
+MongoDB\BSON\Binary::__set_state()
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$tests = [
+    ['foobar', MongoDB\BSON\Binary::TYPE_GENERIC],
+    ['', MongoDB\BSON\Binary::TYPE_GENERIC],
+    ["\0foo", MongoDB\BSON\Binary::TYPE_GENERIC],
+    [hex2bin('123e4567e89b12d3a456426655440000'), MongoDB\BSON\Binary::TYPE_UUID],
+    [md5('foobar', true), MongoDB\BSON\Binary::TYPE_MD5],
+];
+
+foreach ($tests as $test) {
+    list($data, $type) = $test;
+
+    var_export(MongoDB\BSON\Binary::__set_state([
+        'data' => $data,
+        'type' => $type,
+    ]));
+    echo "\n\n";
+}
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+MongoDB\BSON\Binary::__set_state(array(
+   'data' => 'foobar',
+   'type' => 0,
+))
+
+MongoDB\BSON\Binary::__set_state(array(
+   'data' => '',
+   'type' => 0,
+))
+
+MongoDB\BSON\Binary::__set_state(array(
+   'data' => '' . "\0" . 'foo',
+   'type' => 0,
+))
+
+MongoDB\BSON\Binary::__set_state(array(
+   'data' => '>Eg�ӤVBfUD' . "\0" . '' . "\0" . '',
+   'type' => 4,
+))
+
+MongoDB\BSON\Binary::__set_state(array(
+   'data' => '8X�"0�<�_0fC�?',
```

Comment:
GitHub is not showing `bson-binary-002.phpt` because it contains binary data. The `get_debug_info` handler previously printed the raw data and I opted to do the same with `get_properties`. PHP's documentation mentioned that users should be prepared for serialization data to be a binary string (possibly containing null bytes and other control characters), and I don't see a limitation with `var_export()`. Could this be a problem for `var_dump()` output?

As-is, this makes `__set_state()` consistent with the constructor in accepting raw data. If you would rather change `get_properties` to return hex-encoded data, we can then convert data from hex to binary during `__set_state()` and `__wakeup()`. Let's create a separate ticket for that if desired.